### PR TITLE
Adding test for django 1.11 and django 2.2

### PR DESCRIPTION
**Pull request by vad33r**
- Adding tests for django111 and django22

- Remove eox-tagging from requirements

- Updating setup.py and adding edx-opaque-keys to requirements

- Adding TransactionTestCase to eox-tenant/test

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,35 +1 @@
...
-pbr==5.1.1                # via mock, stevedore
-pycodestyle==2.4.0        # via -r requirements.in
-pylint==1.9.3             # via -r requirements.in
-pymongo==3.7.2            # via edx-opaque-keys
-pytz==2018.7              # via django
-singledispatch==3.4.0.3   # via astroid, pylint
-six==1.11.0               # via astroid, edx-opaque-keys, mock, pylint, singledispatch, stevedore
-stevedore==1.30.0         # via edx-opaque-keys
-wrapt==1.10.11            # via astroid
+-r requirements/base.txt
```
Comment (andrey-canon) [inline requirements.txt:1]: why do you keep this file ?

File: requirements/base.in [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# Main requirements of your plugin application.
+-c constraints.txt
+
+Django
+djangorestframework
+django-mysql
```
Comment (andrey-canon) [inline requirements/base.in:6]: is this a requirement ?

File: requirements/test.in [hunk 3]
```diff
@@ -0,0 +1,13 @@
+-c constraints.txt
+-r base.txt
+
+django-crum
```
Comment (andrey-canon) [inline requirements/test.in:4]: django-crum is not for testing

File: requirements/test.in [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+ddt
+pycodestyle
+pylint
+coverage
+mock
+testfixtures
+path.py
+django-fake-model
+edx-opaque-keys[django]
```
Comment (andrey-canon) [inline requirements/test.in:13]: This is already in base.in

**Review by andrey-canon — CHANGES_REQUESTED** [review]

File: requirements/test.in [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+
+django-crum
+ddt
+pycodestyle
+pylint
+coverage
+mock
+testfixtures
+path.py
+django-fake-model
```
Comment (mariajgrimaldi) [inline requirements/test.in:12]: is this used?

Comment (vad33r) [reply requirements/test.in]: i fixed it

File: requirements/test.in (hunk 4 above)
Comment (vad33r) [reply requirements/test.in]: i fixed it

File: requirements/test.in (hunk 3 above)
Comment (vad33r) [reply requirements/test.in]: oks, i fixed it

File: requirements/base.in (hunk 2 above)
Comment (vad33r) [reply requirements/base.in:6]: it was a requirement in the old requirement file

File: requirements.txt (hunk 1 above)
Comment (vad33r) [reply requirements.txt:1]: should i delete it? it's used in the Makefile in make requirements

File: eox_tenant/edxapp_wrapper/backends/branding_api_h_v1.py [hunk 6]
```diff
@@ -1,5 +1,8 @@
 """ backend """
-from branding import api as branding_api  # pylint: disable=import-error
+try:
+    from branding import api as branding_api  # pylint: disable=import-error
+except ImportError:
+    branding_api = object
```
Comment (morenol) [inline eox_tenant/edxapp_wrapper/backends/branding_api_h_v1.py:5]: In which case this is throwing ImportError? Shouldn't we create a new backend for the tests? or for juniper if needed?

File: requirements.txt (hunk 1 above)
Comment (morenol) [reply requirements.txt:1]: Yes, you can delete this and change the Mafefile requirements target to use the new requirements file (base.txt)

File: requirements/base.in (hunk 2 above)
Comment (morenol) [reply requirements/base.in:6]: I think that yo should add a marker here:

"django-mysql>=3.6; python_version=='3.5'", just like it is currently in setup.py, and the modify setup.py to get the requirements from this file.

Comment (andrey-canon) [reply requirements/base.in:6]: This pr https://github.com/eduNEXT/eox-tenant/pull/103  removed the django-mysql, so  do we need still here ??

Comment (morenol) [reply requirements/base.in:6]: I wasn't aware of that. That's great, I guess that we don't need that anymore.

Comment (vad33r) [reply requirements/base.in]: i removed the requirement

File: requirements.txt (hunk 1 above)
Comment (vad33r) [reply requirements.txt]: i removed the file

File: eox_tenant/edxapp_wrapper/backends/branding_api_h_v1.py (hunk 6 above)
Comment (andrey-canon) [reply eox_tenant/edxapp_wrapper/backends/branding_api_h_v1.py:5]: This is just for tests, when we added new settings a file which imports from this is loaded

**Review by andrey-canon — APPROVED** [review]

**Pull request merged**